### ISSUE TITLE
Clarification on extract_request and extract_response

### ISSUE THREAD

**Issue by M3m3M4n** [aflnet/aflnet#63]
I want to add new protocol to aflnet. Looking through the code, E.G extract_response_codes_http, parser continues after getting status code. Does this mean it is expected that there are multiple requests in buf? Or is this just a guard for mutated input? Same question for extract_requests_x.

**Comment by M3m3M4n** [comment]
As I understand it in i#16 these function is only used in corpus parsing. So to handle a binary protocol with no clear distinction between requests / responds I can just separate individual request into their own file instead of using the pcap dump directly, is this correct?

**Comment by thuanpv** [comment]
Hi @M3m3M4n. Regarding your first question, my answer is yes, the extract_response_codes_XXX function accepts the whole response from the server under test for a particular sequence of requests and it returns a sequence of response codes --- not a single response code. Regarding your second question, the function extract_requests_XXX is used for corpus parsing while the extract_response_codes_XXX function is used to support state machine inference. For any protocol, as long as you can form sequences of client requests,  e.g., from captured network traffic, and use them as seeds + extract server responses, AFLNet should work. For protocols in which the response codes are not explicit, you can somehow extract some part(s) of the responses, encode them, e.g., using a simple hash function, and use those encoded values as response codes. An example is the TLS/DTLS protocol that is supported by AFLNet. I hope it helps. Thuan

**Comment by M3m3M4n** [comment]
Hi @thuanpv. Thank you for answering. I'm also trying to implement a complete blackbox mode for remote target, it should be like enabling dumb_mode + no_forkserver and managed (optionally) by extra scripts.

The plan so far is modifying send_over_network() (and others) to report crash/error in recv step after sending message. Offloading from run_target() since target cannot be managed locally anymore. This might not be stable depends on network connection.

I wonder if this affect your message generation algorithm, aside from no coverage info (dumb_mode)?

**Comment by thuanpv** [comment]
No, I don't think it will affect the way AFLNet mutate message sequences to generate new ones.
